src/browser/setupWorker/setupWorker.ts

```diff
@@ -36,6 +37,8 @@ export class SetupWorkerApi
   private startHandler: StartHandler = null as any
   private stopHandler: StopHandler = null as any
   private listeners: Array<Listener>
+  private workerActivationPromise: DeferredPromise<ServiceWorkerRegistration> | null =
```

Comment:
Since the activation promise can be rejected now, it fulfills to either the registration reference or `null`.